### DISCUSSION TITLE
HTTPS and user auth

### DISCUSSION THREAD

**Discussion by tedraymond** [imartinez/privateGPT#1362] [Ideas]
Anyone try implementing https and user authentication via local db or ldap? Curious if gradio cann easily support or would it be better to run behind nginx or other web platform.

**Comment by BrunoBosshard** [comment]
I use nginx as https front-end for Gradio and it works well. I don't see a problem with user authentication with nginx, I don't think that you can do what you want with Gradio alone.

**Reply by tedraymond** [reply]
Thanks for the insight. I'll kind of lean that direction.

**Reply by tedraymond** [reply]
Accomplished in the uvicorn __main__.py

**Reply by ne-ak** [reply]
@tedraymond Could you explain how did you accomplish it? I installed mkcert for making certificates. Then added following to main.py but both Edge and Firefox prevents the access. I see this "Warning: Potential Security Risk Ahead" 
uvicorn.run(app,
                host="0.0.0.0",
                port=8001,
                ssl_keyfile="./localhost+2-key.pem",
                ssl_certfile="./localhost+2.pem"
                )

**Comment by tedraymond** [comment]
Yes , that will happen with self signed certificates. On your local browser
you should be able to install the certificate to bypass that or you can use
a public signing authority to create “real” certificates.

**Comment by wesleyscaldwell** [comment]
This is not directly relevant, but might be. 

Has anyone tried configuring a reverse proxy to handle URL prefixes for privategpt? For instance, directing traffic from https://example.com/myapp/ to the backend without hardcoding the slug in the application code. 
Any experiences or tips on using headers like X-Forwarded-Prefix to dynamically manage these paths would be helpful.

I have the UI rendering, but there is a "connection errored out" message.

**Reply by hectorferronato** [reply]
Im getting the same error. Did you figure out? Seems like the frontend keeps making a call to the original PORT and the CORS does not work properly when using nginx
